(DOCSP-13195): Stitch->Realm Android Migration (#647)
mongodb/docs-realm: source/android/migrate.txt

.. _android-migrating-from-stitch-to-realm:

=====================================

> Stitch SDK to the Realm SDK Migration

Reviewer: Can we alter this title to be a verb/action phrase:

`Migrate the Stitch SDK to the Realm SDK` or `Migrate Your Stitch App to a MongoDB Realm App` or `Migrate Your Android App from Stitch to Realm` (or something along those lines)

Author: There's a spot of trouble there -- I actually made this the title to keep the title fitting reasonably (in only two lines, lol) on the sidebar. Maybe just "Upgrade from Stitch to Realm" ?

Author: Very open to brief alternatives!

=====================================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------
If you have an existing app built with the Stitch SDKs, you should migrate your 
app to use the new Realm SDKs. While much of the logic and flow of information 
hasn't changed, there are a few important changes in the way your app connects to 
the {+realm+} backend.

New Features
------------

> - Android applications written in Kotlin are now supported by the Realm Android SDK.

Reviewer: ```suggestion
- Android applications written in Kotlin are now supported by the Realm Android SDK.
```

minor nit, to avoid passive voice

Author: I don't see a difference between the original line and your suggestion? Copy/paste issue?

Author: But rewrote anyway to active voice, thanks!

> - The Realm Android SDK now includes {+client-database+}, which includes local object storage and the ability to synchronize document changes across devices over the internet.

Reviewer: ```suggestion
- The Realm Android SDK now includes {+client-database+}, which includes local object storage and synchronizing document changes across devices over the internet.
```

The above suggestion accomplishes the same thing but is slightly less wordy

Reviewer: you might even link to the [sync](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/whatschanged/sync) docs here

Deprecated Features
-------------------

- The {+realm+} SDKs do not provide hooks for calling Services (like the 
  :doc:`Twilio <services/twilio>` and :doc:`AWS <services/aws>` services). We 
  recommend using the AWS and/or Twilio ``npm`` packages within {+realm+} 
  functions to connect to these services moving forward. For more information, 
  see :doc:`Upload External Dependencies </functions/upload-external-dependencies>`.

Changes
-------

.. list-table::
   :header-rows: 1
   :widths: 50 50

   * - What's Changed?

>      - Migration

Reviewer: **Not a required change or suggestion:**
Should the second column (Migration) be more descriptive? Perhaps: `Migration Recommendations` or `How to Migrate` or something.

Author: Changing to "solution"

   
   * - The backend app connections previously accessed via `StitchAppClient <http://stitch-sdks.s3-website-us-east-1.amazonaws.com/stitch-sdks/java/4/com/mongodb/stitch/android/core/StitchAppClient.html>`__ are now accessed through :java-sdk:`App <io/realm/mongodb/App.html>`.

>      - Migrate all occurrences of ``StitchAppClient`` to ``App``.

Reviewer: Would it be helpful to link to the Android SDK for all the API methods in the rows under "Migrate". So for instance the sentence above might be rewritten:

- Migrate all occurrences of [StitchAppClient](http://stitch-sdks.s3-website-us-east-1.amazonaws.com/stitch-sdks/java/4/com/mongodb/stitch/android/core/StitchAppClient.html) to [App](https://docs.mongodb.com/realm-sdks/java/10.2.0/index.html)

Author: It's already linked on the left side, which has deeper information about the changes anyway. Do you think that an extra link would help users? I'm currently thinking of the right side as a "simple explanation + for more information, visit" so i could go either way.

Reviewer: Ah no you're right, keep it as is! I missed the links on the leftside since I was reviewing it through github line by line.

   * - Application users previously accessed via `StitchUser <http://stitch-sdks.s3-website-us-east-1.amazonaws.com/stitch-sdks/java/4/com/mongodb/stitch/android/core/auth/StitchUser.html>`__ are now accessed through :java-sdk:`User <io/realm/mongodb/User.html>`.
     - Migrate all occurrences of ``StitchUser`` to ``User``.

>    * - In the StitchSDK, network requests like `callFunction() <http://stitch-sdks.s3-website-us-east-1.amazonaws.com/stitch-sdks/java/4/com/mongodb/stitch/android/core/StitchAppClient.html#callFunction(java.lang.String,java.util.List)>`__ were asynchronous by default. In the Realm Android SDK, you can choose synchronous or asynchronous requests with the "Async" suffix, e.g. :java-sdk:`callFunction() <io/realm/mongodb/functions/Functions.html#callFunction-java.lang.String-java.util.List-->` and :java-sdk:`callFunctionAsync() <io/realm/mongodb/functions/Functions.html#callFunctionAsync-java.lang.String-java.util.List--io.realm.mongodb.App.Callback->`.

Reviewer: ```suggestion
   * - In the Stitch SDK, network requests like `callFunction() <http://stitch-sdks.s3-website-us-east-1.amazonaws.com/stitch-sdks/java/4/com/mongodb/stitch/android/core/StitchAppClient.html#callFunction(java.lang.String,java.util.List)>`__ were asynchronous by default. In the Realm Android SDK, you can choose synchronous or asynchronous requests with the "Async" suffix, e.g. :java-sdk:`callFunction() <io/realm/mongodb/functions/Functions.html#callFunction-java.lang.String-java.util.List-->` and :java-sdk:`callFunctionAsync() <io/realm/mongodb/functions/Functions.html#callFunctionAsync-java.lang.String-java.util.List--io.realm.mongodb.App.Callback->`.
```

Reviewer: Typo/required change ^
